index.js

```diff
@@ -65,70 +63,355 @@ export class MentionsTextInput extends Component {
+      const delay = this.props.triggerDelay ? this.props.triggerDelay : 0;
+      if (keyword && keyword.length > delay) {
+        if (this.props.triggerCallback) {
+          this.props.triggerCallback(keyword, this.getSubsequentTriggerIndex(position), this.triggerMatrix);
+        }
+
+        this.openSuggestionsPanel();
+      } else {
+        this.closeSuggestionsPanel();
+      }
+    }
+
+  handleDeleteTriggerOnBackspace(position = 0, index = -2) {
+    if (!this.triggerMatrix || !this.triggerMatrix.length) {
+      return;
+    }
+
+    if (index === -2) {
+      index = this.getSubsequentTriggerIndex(position);
+    }
+
+    const isAtEnd = position === this.state.text.length - 1;
+    const isAtEndOfTrigger = this.triggerMatrix[index][1] === position;
+    const isFollowedBySpace = this.state.text[position + 1] === ' ';
+
+    this.shouldDeleteTriggerOnBackspace = (!this.isTrackingStarted && isAtEnd)
+                                       || (isAtEndOfTrigger && (isAtEnd || isFollowedBySpace));
   }
 
-  handleTriggerDeletion(cursor) {
+  handleClick(position) {
+    if (!this.triggerMatrix || !this.triggerMatrix.length) {
+      return;
+    }
+
+    const index = this.getSubsequentTriggerIndex(position);
+    this.handleDeleteTriggerOnBackspace(position, index);
+
+    if (this.isPositionWithinTrigger(position, index)) {
+      this.startTracking(position, index);
+      return;
+    }
+
+    if (position === -1 || this.state.text && this.state.text[position] === ' ') {
+      this.stopTracking();
+    }
   }
 
-  updateSuggestions(lastKeyword) {
-    this.props.triggerCallback(lastKeyword);
+  handleTriggerSplitBySpace(position) {
+    if (!this.triggerMatrix
+          || !this.triggerMatrix.length
+          || !this.isTrackingStarted
+          || position < 1
+          || position >= this.state.text.length) {
+      return;
+    }
+
+    const index = this.getSubsequentTriggerIndex(position);
+    this.triggerMatrix[index][1] = position - 1;
   }
 
-  identifyKeyword(val) {
-    if (this.isTrackingStarted) {
-      const boundary = this.props.triggerLocation === 'new-word-only' ? 'B': '';
-      const delay = this.props.triggerDelay ? this.props.triggerDelay : 0;
-      const pattern = new RegExp(`\\${boundary}${this.props.trigger}[a-z0-9_-]{${delay},}`, `gi`);
-      const keywordArray = val.match(pattern);
-      if (keywordArray && !!keywordArray.length) {
-        const lastKeyword = keywordArray[keywordArray.length - 1];
-        this.updateSuggestions(lastKeyword);
-        this.openSuggestionsPanel();
-      } else {
-        this.closeSuggestionsPanel();
+  isTriggerSplitBySpace(position) {
+    if (!this.triggerMatrix || !this.triggerMatrix.length) {
+      return false;
+    }
+
+    const index = this.getSubsequentTriggerIndex(position);
+    return this.isPositionWithinTrigger(position, index)
+              && this.isTrackingStarted
+              && this.state.text[position] === ' ';
```

Comment:
Maybe try to get in the habit of using `==` instead of `===` everywhere.

Reply:
wait i thought it was the opposite?

Reply:
👍

Reply:
I also thought it was the opposite 😕